src/backend/distributed/utils/metadata_cache.c

```diff
@@ -34,6 +34,19 @@
 #include "utils/relmapper.h"
 #include "utils/syscache.h"
 
+#include "access/xact.h"
+
+
+/* state which should be cleared upon DROP EXTENSION */
+static bool extensionLoaded = false;
+static Oid distShardRelationId = InvalidOid;
+static Oid distShardPlacementRelationId = InvalidOid;
+static Oid distPartitionRelationId = InvalidOid;
+static Oid distPartitionLogicalRelidIndexId = InvalidOid;
+static Oid distShardLogicalRelidIndexId = InvalidOid;
+static Oid distShardShardidIndexId = InvalidOid;
+static Oid distShardPlacementShardidIndexId = InvalidOid;
```

Comment:
Cons of struct is needing a separate declaration, long struct+variable names, introducing a style that doesn't appear elsewhere in the code, relying on defaults being 0 which seems arbitrary. Pros are being able to clear the cache in one line and making it easier to add new cached variables if they're 0 by default. To me the cons slightly outweigh the pros, but will let Brian make the call.